src/main/java/DukeCommandValidator.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    public static boolean validateTodo(String input){
+        if(input.length()>=MINIMUN_TODO_LENGTH) {
+            if (input.substring(0, 5).equals("todo ")) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static boolean validateDeadline(String input){
+        if(input.length()<MINIMUM_DEADLINE_LENGTH){
+            return false;
+        }
+        if (!input.substring(0, 9).equals("deadline ")){
+            return false;
+        }
+        // validate substring before backslash
+        int indexOfBackslash = input.indexOf("/");
+        if(indexOfBackslash < 11){
+            return false;
+        }
+        // validate substring after backslash
+        String stringAfterBackslash = input.substring(indexOfBackslash+1);
+        int indexFirstSpaceAfterBackslash = stringAfterBackslash.indexOf(" ");
+        if(indexFirstSpaceAfterBackslash < 1){
+            return false;
+        }
+        if(stringAfterBackslash.length() == indexFirstSpaceAfterBackslash + 1){
+            return false;
+        }
+        return true;
+    }
+
+    public static boolean validateEvent(String input){
+        if(input.length()<MINIMUM_EVENT_LENGTH){
+            return false;
+        }
+        if (!input.substring(0, 6).equals("event ")){
+            return false;
+        }
+        // validate substring before backslash
+        int indexOfBackslash = input.indexOf("/");
+        if(indexOfBackslash < 8){
+            return false;
+        }
+        // validate substring after backslash
+        String stringAfterBackslash = input.substring(indexOfBackslash+1);
+        int indexFirstSpaceAfterBackslash = stringAfterBackslash.indexOf(" ");
+        if(indexFirstSpaceAfterBackslash < 1){
+            return false;
+        }
+        if(stringAfterBackslash.length() == indexFirstSpaceAfterBackslash + 1){
+            return false;
+        }
+        return true;
+    }
+
+
+    public static Boolean validateAddCommand(String input){
+
+        if(input.length()>=MINIMUN_TODO_LENGTH) {
+            if (input.substring(0, 5).equals("todo ")) {
+                return true;
+            }
+        }
+
+        if(input.length()>=MINIMUM_EVENT_LENGTH) {
+            if (input.substring(0, 6).equals("event ")) {
+                return true;
+            }
+        }
+
+        if(input.length()>=MINIMUM_DEADLINE_LENGTH) {
+            if (input.substring(0, 9).equals("deadline ")) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public static boolean validateDoneCommand(String input){
+        if(input.length()<=MINIMUM_DONE_LENGTH){
+            return false;
+        }
+        String firstFiveChars = input.substring(0, 5);
+        String sixthToLastChars = input.substring(5);
+        if(firstFiveChars.equals("done ") && isInteger(sixthToLastChars)){
+            return true;
+        }
+        return false;
+    }
```

Comment:
Would be better if the spacings for brackets and parenthesis are more consistent. Can consider using the auto-formatting function for code style in your ide